From 830ff1e01d498407065be3742453d38c70636bdb Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 30 Sep 2021 16:30:54 +0300
Subject: [PATCH] [SW-57783] pci: remove memory access protection during iatu init

Testing shows that some hosts misbehave when the device pulls down
the mem_en bit right after the host sets it up.
Since now we are protecting each region of the iatu during
initialization, we can remove the memory protection by the mem_en
bit of the configuration space.

Change-Id: I217ad976e53ca362ab467dd4ada1663512df9d7c
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/asic.c b/arch/gaudi/arm/asic.c
index e191143..f053a3e 100644
--- a/arch/gaudi/arm/asic.c
+++ b/arch/gaudi/arm/asic.c
@@ -383,8 +383,6 @@
 	int i;
 	struct hl_pci_outbound_region outbound_region;
 
-	hl_pcie_disable_mem_access();
-
 	for (i = 0; i < ARRAY_SIZE(gaudi_pci_inbound_regions); ++i)
 		hl_pci_set_inbound_region(i, &gaudi_pci_inbound_regions[i]);
 
@@ -392,8 +390,6 @@
 	outbound_region.addr = HOST_PHYS_BASE;
 	outbound_region.size = HOST_PHYS_SIZE;
 	hl_pci_set_outbound_region(&outbound_region);
-
-	hl_pcie_enable_mem_access();
 }
 
 void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 8097d60..61c620c 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -169,8 +169,6 @@
 	int i;
 	struct hl_pci_outbound_region outbound_region;
 
-	hl_pcie_disable_mem_access();
-
 	for (i = 0; i < ARRAY_SIZE(gaudi2_pci_inbound_regions); ++i)
 		hl_pci_set_inbound_region(i, &gaudi2_pci_inbound_regions[i]);
 
@@ -178,6 +176,4 @@
 	outbound_region.addr = HOST_PHYS_BASE_0;
 	outbound_region.size = HOST_PHYS_SIZE_0;
 	hl_pci_set_outbound_region(&outbound_region);
-
-	hl_pcie_enable_mem_access();
 }
diff --git a/arch/greco/common/asic.c b/arch/greco/common/asic.c
index e1a959d..1f85442 100644
--- a/arch/greco/common/asic.c
+++ b/arch/greco/common/asic.c
@@ -60,8 +60,6 @@
 	int i;
 	struct hl_pci_outbound_region outbound_region;
 
-	hl_pcie_disable_mem_access();
-
 	for (i = 0; i < ARRAY_SIZE(greco_pci_inbound_regions); ++i)
 		hl_pci_set_inbound_region(i, &greco_pci_inbound_regions[i]);
 
@@ -69,8 +67,6 @@
 	outbound_region.addr = HOST_PHYS_BASE;
 	outbound_region.size = HOST_PHYS_SIZE;
 	hl_pci_set_outbound_region(&outbound_region);
-
-	hl_pcie_enable_mem_access();
 }
 
 void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
